js/usercss.js

```diff
@@ -0,0 +1,503 @@
+          section.code = varDef + section.code;
+        }
+      }
+    },
+    stylus: {
+      preprocess(source, vars) {
+        return loadScript('/vendor/stylus-lang/stylus.min.js').then(() => (
+          new Promise((resolve, reject) => {
+            let varDef = '';
+            for (const key of Object.keys(vars)) {
+              varDef += `${key} = ${vars[key].value};\n`;
+            }
+
+            // eslint-disable-next-line no-undef
+            stylus(varDef + source).render((err, output) => {
+              if (err) {
+                reject(err);
+              } else {
+                resolve(output);
+              }
+            });
+          })
+        ));
+      }
+    },
+    uso: {
+      preprocess(source, vars) {
+        const pool = new Map();
+        return Promise.resolve(doReplace(source));
+
+        function getValue(name, rgb) {
+          if (!vars.hasOwnProperty(name)) {
+            if (name.endsWith('-rgb')) {
+              return getValue(name.slice(0, -4), true);
+            }
+            return null;
+          }
+          if (rgb) {
+            if (vars[name].type === 'color') {
+              // eslint-disable-next-line no-use-before-define
+              const color = colorParser.parse(vars[name].value);
+              return `${color.r}, ${color.g}, ${color.b}`;
+            }
+            return null;
+          }
+          if (vars[name].type === 'dropdown' || vars[name].type === 'select') {
+            // prevent infinite recursion
+            pool.set(name, '');
+            return doReplace(vars[name].value);
+          }
+          return vars[name].value;
+        }
+
+        function doReplace(text) {
+          return text.replace(/\/\*\[\[([\w-]+)\]\]\*\//g, (match, name) => {
+            if (!pool.has(name)) {
+              const value = getValue(name);
+              pool.set(name, value === null ? match : value);
+            }
+            return pool.get(name);
+          });
+        }
+      }
+    }
+  };
+
+  function getMetaSource(source) {
+    const commentRe = /\/\*[\s\S]*?\*\//g;
+    const metaRe = /==userstyle==[\s\S]*?==\/userstyle==/i;
+
+    let m;
+    // iterate through each comment
+    while ((m = commentRe.exec(source))) {
+      const commentSource = source.slice(m.index, m.index + m[0].length);
+      const n = commentSource.match(metaRe);
+      if (n) {
+        return n[0];
+      }
+    }
+  }
+
+  function buildMeta(source) {
+    const style = _buildMeta(source);
+    validate(style);
+    return style;
+  }
+
+  function parseWord(state, error = 'invalid word') {
+    const match = state.text.slice(state.re.lastIndex).match(/^([\w-]+)\s*/);
+    if (!match) {
+      throw new Error(error);
+    }
+    state.value = match[1];
+    state.re.lastIndex += match[0].length;
+  }
+
+  function parseVar(state) {
+    const result = {
+      type: null,
+      label: null,
+      name: null,
+      value: null,
+      default: null,
```

Comment:
Same here.

Reply:
So we change `result.default` to `result.defaultValue`?